addon-test-support/@ember/test-helpers/wait-until.js

```diff
@@ -0,0 +1,22 @@
+import { Promise } from 'rsvp';
+
+export default function(callback, options = {}) {
+  let timeout = 'timeout' in options ? options.timeout : 1000;
+  let waitUntilTimedOut = new Error('waitUntil timed out');
+
+  return new Promise(function(resolve, reject) {
+    let time = 0;
+    function tick() {
+      time += 10;
+      let value = callback();
+      if (value) {
+        resolve(value);
+      } else if (time < timeout) {
+        setTimeout(tick, 10);
+      } else {
+        reject(waitUntilTimedOut);
+      }
+    }
+    setTimeout(tick, 10);
```

Comment:
should this call `tick()` directly?

Reply:
I did think about that, but doing that would allow it to resolve sync if wrapped in a run-loop, and we _never_ want to resolve sync.